### PR TITLE
[project-base] - add hover timeout to horizontal menu

### PR DESCRIPTION
**Pull request by TomasGottvald**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Add hover in / hover out timeout for horizontal menu. 
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docs/frontend/hover-intent.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# hoverIntent Component
+This JS component is used for add timeout for hover over and hover out events.
```
Comment (DavidOstrozlik) [inline docs/frontend/hover-intent.md]:
```suggestion
This JS component is used to add timeout for hover over and hover out events.
```

File: docs/frontend/hover-intent.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+This component uses hoverIntent plugin (http://briancherne.github.io/jquery-hoverIntent/).
```
Comment (DavidOstrozlik) [inline docs/frontend/hover-intent.md]:
```suggestion
This component uses [hoverIntent plugin](http://briancherne.github.io/jquery-hoverIntent/) from jQuery.
```

**Comment by TomasGottvald** [comment]
> 
> 
> Please check if the jquery plugin is the same one as in admin and if so , check if you can use the same plugin so we don't need to duplicate it.

Yes, there are the same. I used newer one in version 1.10. In SSFW we have 1.8. version. So we can use newer one in admin too. But I tried to find any way to use it from admin, but finally I think it is better to have frontend and backend separately.

File: docs/frontend/hover-intent.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# hoverIntent Component
```
Comment (TomasLudvik) [inline docs/frontend/hover-intent.md]: This should start with big letter and also name of the file should be same as first title

File: docs/frontend/hover-intent.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+This plugin is binded to elements with class `.js-hover-intent`
+
+### Force click
+In some scenarios you need to add hover functionality and simulate click on enother element. You can define it by `data-hover-intent-force-click="true"` and data element with `data-hover-intent-force-click-element=".js-element-to-click-class"`. I makes hover effect with defined interval a calls click event on element with class `.js-element-to-click-class`.
+
+### Javascript default settings
+```javascript
+var interval = 300;
+var timeout = 300;
```
Comment (TomasLudvik) [inline docs/frontend/hover-intent.md]: This is not same as is in hoverIntent.js

File: docs/frontend/hover-intent.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+var forceClick = false;
+var forceClickElement = '';
+var linkOnMobile = false;
+```
+
+### Html element settings
+```html
+<div class="js-hover-intent"
+data-hover-intent-interval="300"
+data-hover-intent-timeout="300"
```
Comment (TomasLudvik) [inline docs/frontend/hover-intent.md]: This is not same as is in hoverIntent.js

File: upgrade/UPGRADE-v9.0.0-dev.md [hunk 6]
```diff
@@ -302,9 +302,32 @@ There you can find links to upgrade notes for other versions too.
...
+- add hover timeout to horizontal menu ([#1564](https://github.com/shopsys/shopsys/pull/1564))
+    - you can skip this task if you have your custom design
+    - move loader to hidden submenu - so it can not interrupt hover `src/Resources/scripts/frontend/categoryPanel.js`
+        ```diff
+          function loadCategoryItemContent ($categoryItem, url) {
+              Shopsys.ajax({
+        -          loaderElement: $categoryItem,
+        +          loaderElement: $categoryItem.find('.js-category-list-placeholder'),
+        ```
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v9.0.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v9.0.0-dev.md [hunk 7]
```diff
@@ -302,9 +302,32 @@ There you can find links to upgrade notes for other versions too.
              - public function __construct(string $entityName, int $entityId, string $type, string $temporaryFilename, int $position)
              + public function __construct(string $entityName, int $entityId, string $type, string $temporaryFilename, string $uploadedFilename, int $position)
             ```
-          
+
  - There is a new base html layout with horizontal menu and product filter placed in left panel, for detail information see [the separate article](upgrade-instructions-for-base-layout.md)
- 
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v9.0.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v9.0.0-dev.md [hunk 8]
```diff
@@ -302,9 +302,32 @@ There you can find links to upgrade notes for other versions too.
...
+    - move loader to hidden submenu - so it can not interrupt hover `src/Resources/scripts/frontend/categoryPanel.js`
+        ```diff
+          function loadCategoryItemContent ($categoryItem, url) {
+              Shopsys.ajax({
+        -          loaderElement: $categoryItem,
+        +          loaderElement: $categoryItem.find('.js-category-list-placeholder'),
+        ```
+
+    - add new js plugin hoverIntent v1.10.1 `src/Resources/scripts/frontend/plugins/jquery.hoverIntent.js` (https://github.com/shopsys/shopsys/tree/master/project-base/src/Resources/scripts/frontend/plugins/jquery.hoverIntent.js)
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v9.0.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v9.0.0-dev.md [hunk 9]
```diff
@@ -302,9 +302,32 @@ There you can find links to upgrade notes for other versions too.
...
+    - add new js component `src/Resources/scripts/frontend/components/hoverIntent.js` (https://github.com/shopsys/shopsys/tree/master/project-base/src/Resources/scripts/frontend/components/hoverIntent.js)
+
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v9.0.0-dev.md]:
```suggestion
```

File: upgrade/UPGRADE-v9.0.0-dev.md [hunk 10]
```diff
@@ -302,9 +302,32 @@ There you can find links to upgrade notes for other versions too.
...
+    - add classes to hover menu `templates/Front/Content/Category/panel.html.twig`
```
Comment (TomasLudvik) [inline upgrade/UPGRADE-v9.0.0-dev.md]:
```suggestion
    - add class and data attributes to hover menu `templates/Front/Content/Category/panel.html.twig`
```

File: docs/frontend/hover-intent.md (hunk 3 above)
Comment (TomasGottvald) [reply docs/frontend/hover-intent.md]: fixed

File: docs/frontend/hover-intent.md (hunk 4 above)
Comment (TomasGottvald) [reply docs/frontend/hover-intent.md]: Fixed according to final settings

File: docs/frontend/hover-intent.md (hunk 5 above)
Comment (TomasGottvald) [reply docs/frontend/hover-intent.md]: fixed according to final settings

File: upgrade/UPGRADE-v9.0.0-dev.md (hunk 10 above)
Comment (TomasGottvald) [reply upgrade/UPGRADE-v9.0.0-dev.md]: fixed, thx

File: project-base/src/Resources/scripts/frontend/categoryPanel.js [hunk 11]
```diff
@@ -17,22 +17,27 @@
...
 
-        if (isOpen) {
-            $categoryList.slideUp('fast');
-            $categoryItem.removeClass('active');
-        } else if ($categoryList.length > 0) {
-            $categoryList.slideDown('fast');
-            $categoryItem.addClass('active');
-        } else {
-            loadCategoryItemContent($categoryItem, $categoryCollapseControl.data('url'));
+        if(!$categoryItem.hasClass("intented")){
```
Comment (TomasLudvik) [inline project-base/src/Resources/scripts/frontend/categoryPanel.js]:
```suggestion
        if (!$categoryItem.hasClass("intented")) {
```

**Pull request merged**